### ISSUE TITLE
[276 Week1]关于第一周的一些感受，以及对于PriorityQueue源码的理解

### ISSUE THREAD

**Issue by ydbB** [algorithm004-01/algorithm004-01#219]
这周学习了关于move-zeros，学会了数组操作的技巧，同时看了国际站上将移动0假想为滚雪球的操作，深深吸引了自己，同时也学到了关于使用栈的类型，最近重复的想法，特别是对于利用栈来解决最大面积的问题，以及利用栈解决储存水的问题，还不是很懂，需要多看。
还有就是深深感到重新做题的重要性，周一做完的题，到现在以及忘得差不多了。
对于PriorityQueue的源码分析：(Queue是接口，其中基本没有内容，不进行分析)
1.PriorityQueue需要两个变量，一个是Collection,另一个是Comparator比较器；
2.底层实现也是用的数组；
3.offer可能抛出NPL，返回boolean值，poll可能会返回null；
4.add返回boolean值；remove可能抛出NPL
其他常规的扩容与其他Queue类似。

**Comment by tigerzhang08** [comment]
很多题型确实需要多看几遍，储存水的问题也没看太明白。五毒神掌还是要用起来，加油！

**Comment by Leith14** [comment]
一起加油咯！

**Comment by shniu** [comment]
对 `PriorityQueue` 的分析，有一些跟人见解，Java 中 `PriorityQueue` 的实现是基于堆的，所以分析的侧重点应该在堆的存储结构和堆的插入删除等过程（这个过程也叫做堆化），我认为这个是 `PriorityQueue` 的精髓

**Comment by showx** [comment]
有点难理解，再看看。
